### DISCUSSION TITLE
Using WalletConnect with Content Security Policy

### DISCUSSION THREAD

**Discussion by roosemberth** [WalletConnect/walletconnect-monorepo#3379] [Developer Support] [answered]
I'm looking to integrate WalletConnect in an existing application to ask the user for an ETH address and then to sign a message with it.

I'm wondering what CSP directives should I use. So far, I've added `connect-src wss://relay.walletconnect.com https://explorer-api.walletconnect.com; frame-src https://verify.walletconnect.com; img-src https://explorer-api.walletconnect.com;`, but this is based in my own testing, as I couldn't find a list in your documentation.

Thanks in advance for your answers!

**Comment by philogicae** [comment]
We need to have a page about CSP in the documentation, please guys

**Reply by mplummeridge** [reply]
<img width="562" alt="image" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/818654/545ab38e-a07c-43a4-b1ba-dff6a025d2bb"> where?

**Comment by apetta** [comment]
+1

Would be great to have this instead of digging around in the console for sources

**Comment by wh173-c47** [comment]
And this is still unanswered... +1..

I mean I don't know... basics security stuffs...

**Comment by glitch-txs** [comment, marked as answer]
This should've been an open issue in the docs repo, sorry I missed the discussion <3

https://github.com/WalletConnect/walletconnect-docs/pull/1603

**Comment by chris13524** [comment]
Docs for CSP here: https://docs.walletconnect.com/advanced/security/content-security-policy
